class.jetpack.php

```diff
@@ -5333,10 +5333,27 @@ public static function is_staging_site() {
 	 */
 	public static function validate_sync_error_idc_option() {
 		$is_valid = false;
-		$sync_error = Jetpack_Options::get_option( 'sync_error_idc' );
+
+		$idc_allowed = get_transient( 'jetpack_idc_allowed' );
+		if ( false === $idc_allowed ) {
+			$response = wp_remote_get( 'https://jetpack.com/is-idc-allowed/' );
+			if ( is_wp_error( $response ) ) {
```

Comment:
I've been watching at this `if else` for a while now and I think it should be the opposite. There's only one case where it's correct to proceed but there are many cases where the request can fail "for some reason" and we want to fallback. For example if `https://jetpack.com/is-idc-allowed/` is at some point a 404, `wp_remote_get` won't return a WP_Error. So maybe:
```php
$response = wp_remote_get( 'https://jetpack.com/is-idc-allowed/' );
$json = json_decode( wp_remote_retrieve_body( $response ) );
if ( 200 === (int) wp_remote_retrieve_response_code( $response ) && ! empty( $json ) ) {
	$idc_allowed = isset( $json->result ) && $json->result ? '1' : '0';
	$transient_duration = HOUR_IN_SECONDS;
} else {
	$idc_allowed = '1';
	$transient_duration = 5 * MINUTE_IN_SECONDS;
}
```